management/test_lifecycle.py

```diff
@@ -0,0 +1,104 @@
+import sys
+from unittest import mock
+
+import pytest
+
+# avoid requiring PyGitHub to run tests
+sys.modules["github"] = mock.MagicMock()
+
+
+@pytest.mark.parametrize(
+    "input_tag,want",
+    [
+        ("compute-openff_mw-300-600", ([300.0, 600.0], "compute-openff")),
+        (
+            "compute-pr393_mw-300-600-900",
+            ([300.0, 600.0, 900.0], "compute-pr393"),
+        ),
+        ("compute-pr393", ([], "compute-pr393")),
+    ],
+)
+def test_parse_tags(input_tag, want):
+    from lifecycle import parse_tags
+
+    got = parse_tags(input_tag)
+    assert got == want
+
+
+@pytest.mark.parametrize(
+    "dsname,dstype,bins,want",
+    [
+        (
+            "OpenFF Sulfur Hessian Training Coverage Supplement v1.0",
```

Comment:
(not blocking) If these tests are taking a few minutes, maybe find a smaller dataset to ping?

Reply:
I'm going to skip this one for now since these aren't run in CI, but I agree in principle, especially if we add CI or end up having to modify this code again soon.